shared/js/initApp.js

```diff
@@ -1,157 +1,116 @@
+        }
+      },
+      backToPreviousLevel: function() {
+        if (lastServerResponse.previousLevelRedirect) {
+          window.location.href = lastServerResponse.previousLevelRedirect;
+        }
+      },
+      showInstructionsWrapper: function(showInstructions) {
+        // Always skip all pre-level popups on share levels or when configured thus
+        if (this.share || appOptions.level.skipInstructionsPopup) {
+          return;
+        }
 
-    var hasVideo = !!appOptions.autoplayVideo;
-    var hasInstructions = !!(appOptions.level.instructions || appOptions.level.aniGifURL);
+        var hasVideo = !!appOptions.autoplayVideo;
+        var hasInstructions = !!(appOptions.level.instructions ||
+        appOptions.level.aniGifURL);
 
-    if (hasVideo) {
-      showVideoDialog(appOptions.autoplayVideo);
-      if (hasInstructions) {
-        $('.video-modal').on('hidden.bs.modal', function () {
+        if (hasVideo) {
+          showVideoDialog(appOptions.autoplayVideo);
+          if (hasInstructions) {
+            $('.video-modal').on('hidden.bs.modal', function() {
+              showInstructions();
+            });
+          }
+        } else if (hasInstructions) {
           showInstructions();
-        });
+        }
       }
-    } else if (hasInstructions) {
-      showInstructions();
-    }
-  }
-};
-$.extend(true, appOptions, baseOptions);
+    };
+    $.extend(true, appOptions, baseOptions);
 
-// Turn string values into functions for keys that begin with 'fn_' (JSON can't contain function definitions)
-// E.g. { fn_example: 'function () { return; }' } becomes { example: function () { return; } }
-(function fixUpFunctions(node) {
-  if (typeof node !== 'object') {
-    return;
-  }
-  for (var i in node) {
-    if (/^fn_/.test(i)) {
-      try {
-        /* jshint ignore:start */
-        node[i.replace(/^fn_/, '')] = eval('(' + node[i] + ')');
-        /* jshint ignore:end */
-      } catch (e) { }
-    } else {
-      fixUpFunctions(node[i]);
-    }
+    // Turn string values into functions for keys that begin with 'fn_' (JSON can't contain function definitions)
+    // E.g. { fn_example: 'function () { return; }' } becomes { example: function () { return; } }
+    (function fixUpFunctions(node) {
+      if (typeof node !== 'object') {
+        return;
+      }
+      for (var i in node) {
+        if (/^fn_/.test(i)) {
+          try {
+            /* jshint ignore:start */
+            node[i.replace(/^fn_/, '')] = eval('(' + node[i] + ')');
+            /* jshint ignore:end */
+          } catch (e) {
+          }
+        } else {
+          fixUpFunctions(node[i]);
+        }
+      }
+    })(appOptions.level);
+  },
+  init: function () {
+    dashboard.project.init();
+    window[appOptions.app + 'Main'](appOptions);
   }
-})(appOptions.level);
-
-function initApp() {
-  dashboard.project.init();
-  window[appOptions.app + 'Main'](appOptions);
-}
```

Comment:
Up to here, the diff for this file should mostly just be indentation changes (and wrapping with the `apps` namespace).